mailbox/plugin/quota-mailing/src/main/java/org/apache/james/mailbox/quota/mailing/QuotaMailingListenerConfiguration.java

```diff
@@ -179,20 +220,25 @@ public static QuotaMailingListenerConfiguration defaultConfiguration() {
         return builder().build();
     }
 
+
+    private final ImmutableMap<QuotaThreshold, RenderingInformation> toRenderingInformation;
     private final QuotaThresholds thresholds;
     private final Duration gracePeriod;
-    private final String bodyTemplate;
-    private final String subjectTemplate;
     private final String name;
 
-    private QuotaMailingListenerConfiguration(QuotaThresholds thresholds, Duration gracePeriod, String bodyTemplate, String subjectTemplate, String name) {
+    private QuotaMailingListenerConfiguration(ImmutableMap<QuotaThreshold, RenderingInformation> toRenderingInformation,
+                                              QuotaThresholds thresholds, Duration gracePeriod, String name) {
+        this.toRenderingInformation = toRenderingInformation;
         this.thresholds = thresholds;
         this.gracePeriod = gracePeriod;
-        this.bodyTemplate = bodyTemplate;
-        this.subjectTemplate = subjectTemplate;
         this.name = name;
     }
 
+    public RenderingInformation getRenderingInformation(QuotaThreshold quotaThreshold) {
+        return Optional.ofNullable(toRenderingInformation.get(quotaThreshold))
+            .orElse(DEFAULT_RENDERING_INFORMATION);
```

Comment:
you manage the default value twice, there's probably something wrong

Reply:
No. You can ommit rendering informations. Or you can omit a part of this information.